crates/hir-def/src/hir.rs

```diff
@@ -175,6 +175,7 @@ pub enum Expr {
         statements: Box<[Statement]>,
         tail: Option<ExprId>,
         label: Option<LabelId>,
+        has_trailing_item_decl: bool,
```

Comment:
Hmm, this feels wrong to me. I think a better approach at fixing this is to not drop item statements on lowering and instead lower them to an empty `Statement::Item` kind for now (we can give it fields once we make actualyl use of that).

Reply:
Alright, I'll change to the suggested approach ... I initially went with this because I thought it was the least invasive approach 🙂